tests/Unit/ControlSystem/Test_UpdateFunctionOfTime.cpp

```diff
@@ -0,0 +1,228 @@
+  const double expr_time = 1.0;
+  const double update_time = 1.5;
+  const double new_expiration_time = 2.0;
+  domain::FunctionsOfTime::register_derived_with_charm();
+
+  // Construct unordered map
+  const std::string pp_name{"TestPiecewisePolynomial"};
+  const std::string quatfot_name{"TestQuatFunctionOfTime"};
+  std::array<DataVector, deriv_order + 1> init_pp_func{
+      {DataVector{3, 0.0}, DataVector{3, 0.0}, DataVector{3, 0.0}}};
+  std::array<DataVector, 1> init_quat_func{DataVector{4, 1.0}};
+
+  domain::FunctionsOfTime::PiecewisePolynomial<deriv_order> expected_pp{
+      t0, init_pp_func, expr_time};
+  domain::FunctionsOfTime::QuaternionFunctionOfTime<deriv_order>
+      expected_quatfot{t0, init_quat_func, init_pp_func, expr_time};
+
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      f_of_t_map{};
+  f_of_t_map[pp_name] = std::make_unique<
+      domain::FunctionsOfTime::PiecewisePolynomial<deriv_order>>(
+      t0, init_pp_func, expr_time);
+  f_of_t_map[quatfot_name] = std::make_unique<
+      domain::FunctionsOfTime::QuaternionFunctionOfTime<deriv_order>>(
+      t0, init_quat_func, init_pp_func, expr_time);
+
+  const DataVector updated_deriv{3, 1.0};
+
+  // Construct mock system and set to Testing phase
+  ActionTesting::MockRuntimeSystem<TestingMetavariables> runsys{
+      {}, {std::move(f_of_t_map)}};
+  ActionTesting::emplace_singleton_component<
+      TestSingleton<TestingMetavariables>>(make_not_null(&runsys),
+                                           ActionTesting::NodeId{0},
+                                           ActionTesting::LocalCoreId{0});
+  ActionTesting::set_phase(make_not_null(&runsys),
+                           TestingMetavariables::Phase::Testing);
+
+  // Update functions of time in global cache with new deriv
+  auto& cache =
+      ActionTesting::cache<TestSingleton<TestingMetavariables>>(runsys, 0_st);
+
+  for (auto& name : {pp_name, quatfot_name}) {
+    Parallel::mutate<
+        domain::Tags::FunctionsOfTime,
+        control_system::Mutations::update_function_of_time<deriv_order>>(
+        cache, name, update_time, updated_deriv, new_expiration_time);
+  }
+
+  // Update expected function of time
+  expected_pp.update(update_time, updated_deriv, new_expiration_time);
+  expected_quatfot.update(update_time, updated_deriv, new_expiration_time);
+
+  // Check that the FunctionsOfTime are what we expected
+  const auto& cache_f_of_t_map =
+      get<domain::Tags::FunctionsOfTime, TestingMetavariables>(cache);
+  const auto& cache_pp = dynamic_cast<
+      const domain::FunctionsOfTime::PiecewisePolynomial<deriv_order>&>(
+      *(cache_f_of_t_map.at(pp_name)));
+  const auto& cache_quatfot = dynamic_cast<
+      const domain::FunctionsOfTime::QuaternionFunctionOfTime<deriv_order>&>(
+      *(cache_f_of_t_map.at(quatfot_name)));
+
+  CHECK(cache_pp == expected_pp);
+  CHECK(cache_quatfot == expected_quatfot);
```

Comment:
Scope this section so you can reuse the variable names when you check again below.